app/src/org/commcare/views/widgets/QuestionWidget.java

```diff
@@ -388,6 +388,8 @@ private SpannableStringBuilder getTextFromPrompt(FormEntryPrompt prompt) {
         SpannableStringBuilder builder = new SpannableStringBuilder();
         if (prompt.getMarkdownText() != null) {
             builder.append(forceMarkdown(prompt.getMarkdownText()));
+        } else if (mPrompt.getLongText() == null) {
```

Comment:
In case the question is required, this won't build the `asterisk`, Since [we are hiding](https://github.com/dimagi/commcare-android/blob/539b3afeda46597d33fc7406f895fbd6a16043dd/app/src/org/commcare/views/widgets/QuestionWidget.java#L441) the label if there is no text.

Reply:
Thanks for catching this. We can just append an empty string to the builder in this case to solve for the asterisk.

Reply:
@shubham1g5 Though it will set the textView with asterisk, but the label will still be hidden.
So, Shall we show label with just the asterisk in case of required question?

Reply:
ahh I think it's fine in that case for the whole quesiotn text to be hidden and to not show anything.